src/Bridge/Twig/Extension/EnumExtension.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Greg0ire\Enum\Bridge\Twig\Extension;
+
+use Greg0ire\Enum\AbstractEnum;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class EnumExtension extends \Twig_Extension
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(TranslatorInterface $translator = null)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFilters()
+    {
+        return [
+            new \Twig_SimpleFilter('enum_label', [$this, 'label']),
+        ];
+    }
+
+    /**
+     * Displays the label corresponding to a specific value of an enumeration.
+     *
+     * @param mixed       $value             Must exist in the enumeration class specified with $class
+     * @param string      $class             The enum class name
+     * @param string|bool $translationDomain The translation domain to use if the translator if available.
+     *                                       string: Use the specified one
+     *                                       null: Use the default one
+     *                                       false: Do not use the translator
+     * @param bool|string $separator         Namespace separator to use with the class prefix.
+     *                                       - true: I want a separator, which implies to prefix with a class.
+     *                                       - 'ø': I want 'ø' as a separator, which implies to prefix with a class.
+     *                                       - false: I don't want any separator, because I don't even need a prefix.
+     *                                       By default, the value is the configured one if the translator
+     *                                       is available and enabled, false otherwise.
+     *
+     * @return string
+     */
+    public function label($value, $class, $translationDomain = null, $separator = null)
+    {
+        // Not specified. We have to guess the default behavior
+        if (is_null($separator)) {
+            $separator = false !== $translationDomain && !is_null($this->translator)
+                ? AbstractEnum::$defaultNamespaceSeparator
+                : false;
+        }
+
+        if (true === $separator) {
```

Comment:
Didn't especially want to force this yoda condition. But when you know the convention to follow, you take used to write it like this.

BTW, follow convention would earns time like this. :stuck_out_tongue: :runner: